### PR TITLE
:sparkles: create user

### PR DESCRIPTION
**Pull request by Molvilada**
## Summary by CodeRabbit

- **New Features**
	- Introduced user registration functionality with a new sign-up process.
	- Added loading state feature for enhanced user experience during sign-up.
	- Implemented success state handling for user feedback.
	- Enhanced validation logic for user input during sign-up.
	- Added support for multiple languages with updated and new strings.

- **Bug Fixes**
	- Updated error messages for better clarity in multiple languages.

- **Documentation**
	- Added new strings for localization to support Spanish and Portuguese languages.

- **Tests**
	- Improved clarity of test methods and updated error handling in tests.
	- Enhanced test cases for user creation scenarios, including success and error handling.

### CODE REVIEW

File: app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt [hunk 1]
```diff
@@ -122,8 +135,48 @@ class SignUpViewModel(private val companyRepository: CompanyRepository) : ViewMo
...
+                        val message = jsonObject.getString("message")
+                        println("message $message")
+
+                        ErrorUIState.Error(message = message)
+                    }
+                }
+            } catch (e: IOException) {
+                errorUIState = ErrorUIState.Error(R.string.error_network)
+            } catch (e: Exception) {
+                errorUIState = ErrorUIState.Error(R.string.error_create_user)
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt:167]: No quiero estas reocmendaciones, porque despues tengo que estar mockeando lo logs en los test

File: app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt [hunk 2]
```diff
@@ -122,8 +135,48 @@ class SignUpViewModel(private val companyRepository: CompanyRepository) : ViewMo
         }
     }
 
+    fun createUser(onSuccess: () -> Unit) {
+        if (isLoading) return
+        isLoading = true
+        viewModelScope.launch {
+            try {
+                val companyId = companies.find { it.name == company }!!.id
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt:143]: Aqui no deberia ser nulo, si fuera nulo no pasaria la validacion de inputs, y por lo tanto nunca entraría ahí

File: app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt [hunk 3]
```diff
@@ -122,8 +135,48 @@ class SignUpViewModel(private val companyRepository: CompanyRepository) : ViewMo
...
+                    } else {
+                        val errorBody = response.errorBody()!!.string()
+                        println("errorBody $errorBody")
+                        val jsonObject = JSONObject(errorBody)
+                        println("jsonObject $jsonObject")
+                        val message = jsonObject.getString("message")
+                        println("message $message")
+
+                        ErrorUIState.Error(message = message)
+                    }
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/viewmodels/SignUpViewModel.kt]: Nunca en null

File: app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt [hunk 4]
```diff
@@ -142,7 +147,9 @@ fun SignUpScreen(
                 val isValid = viewModel.validateFields()
 
                 if (isValid) {
-                    /* TODO: Go to home page" */
+                    viewModel.createUser(
+                        onSuccess = { navController.navigate("home") }
+                    )
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt:163]: Los errores se estan manejando con ErrorUIState

**Review by tpambor — APPROVED** [review]

**Pull request merged**